### PR TITLE
Fix: Win10 symlink require privilege

### PR DESCRIPTION
**Pull request by vunhatchuong**
## Problem
1. Symlink on Windows 10 Powershell required privilege, so we need to use `cmd` to create a symlink with `mklink`. See issue https://github.com/kevincobain2000/gobrew/issues/174

2. `filepath.EvalSymlinks` doesn't work correctly on Windows. A fix is currently being proposed: https://github.com/golang/go/issues/63703

### CODE REVIEW

**Comment by kevincobain2000** [comment]
Thanks 
Please wait for @juev's review

File: gobrew.go
Comment (juev) [inline gobrew.go:273]: We have a file, `gobrew_windows.go`, which houses the code used only in windows.

It seems that you need to create a separate function like `evalSymlinks`, for which `gobrew_unix.go` will have a normal implementation, and in `gobrew_windows.go` will be specific to windows.

And the code will be easier to read.

File: gobrew.go
Comment (juev) [inline gobrew.go:276]: I need a little more time to check what is offered in
https://github.com/golang/go/issues/63703

I don't quite understand why a call is made specifically ps in this case

File: helpers.go
Comment (juev) [inline helpers.go:353]: It is also necessary to remove the checks on the operating system and add a function like `symlink`, which we will implement differently depending on the type of operating system in our file.

Again, because I don't have windows at hand, I need a little more time.

And the question is, why in one case do you call ps, and here you call cmd? Why such a spread?

Comment (vunhatchuong) [reply helpers.go:353]: To my knowledge, there's no easy way to eval a symlink with `cmd`, and I use `cmd` because it doesn't require privilege to create symlinks.

I'll remove the GOOs check, I've never coded cross-platform code before.

Comment (vunhatchuong) [reply helpers.go]: image

**Comment by juev** [comment]
@kevincobain2000 has become much better now, but I still don't like the idea of calling third-party programs from the code.

In addition, Windows has a lot of shell, that is, many launch options and each may have its own problems.

In my opinion, if we have a problem with creating a simlink, we can simply copy the directory from a certain version to the `current` directory.

For example, when it is necessary to switch the version:

1. Remove the directory with the current version from `versions`
2. Move or copy the version from `current` to `versions`
3. Clear the `current` directory
4. Copy the directory from `versions/{version}` to current

If it is not available, we download from the Internet accordingly.

Here you still need to think about how to solve the problem with copying the `bin` directory. But it seems that it is much easier than starting individual processes.

What do you think?

**Comment by kevincobain2000** [comment]
Hi @juev 
Thanks for the review.
I also do not prefer shell invoked as separate exec.

Your suggestion on version switch is good, it can be done natively - all in GO.
But again, it is windows, I didn't intend to support powershell in the first place, people could just `wsl` instead.

Having said all that, if calling a separate `powershell` command, temporarily solves this issue, then it is cool.
The same implementation will never get merged if it was a unix based os, and would code in GO.
Also I have a windows machine now, but hate the idea of `powershell` in the first place. But people want it so be it as dirty as it can get.

**Comment by kevincobain2000** [comment]
If you are ok then I am also ok for this merge. It's not like it is changing anything for unix based os.

**Comment by kevincobain2000** [comment]
Let me know if you are ok with merge @juev

**Review by juev — APPROVED** [review]
Yes, I disagree with the implementation. It could have been implemented much more correctly.
It's not clear if everything works in sl, why use cmd or ps?

But since we hardly change the usual functionality, so be it.

**Comment by kevincobain2000** [comment]
Thanks @juev will merge it after dinner

**Pull request merged**
